enhancements/ingress/admin-configured-ca-bundle.md

```diff
@@ -0,0 +1,243 @@
+
+```yaml
+apiVersion: v1
+data:
+  ca-bundle.crt: |
+    -----BEGIN CERTIFICATE-----
+    <service-ca-bundle content>
+    -----END CERTIFICATE-----
+    -----BEGIN CERTIFICATE-----
+    <admin-ca-bundle content>
+    -----END CERTIFICATE-----    
+kind: ConfigMap
+metadata:
+  name: ingress-ca-bundle
+  namespace: openshift-ingress
+```
+
+OpenShift Ingress operator should create a watcher to observe `admin-ca-bundle` and 
+`service-ca-bundle` for any changes and update `ingress-ca-bundle` accordingly.
+
+OpenShift Router currently uses the CA certificate provided as a process argument
+`default-destination-ca-path` or set as an environment variable `DEFAULT_DESTINATION_CA_PATH`
+as the CA certificate for verifying the service's certificate for reencrypt termination type.
+
+CA certificate bundle is currently made available to the router using the volume mount option 
+at the path `/var/run/configmaps/service-ca/service-ca.crt` which will be updated to 
+`/var/run/configmaps/ca-trust/ca-bundle.crt` with the `ingress-ca-bundle` content.
+
+OpenShift Router should create a file watcher to observe `/var/run/configmaps/ca-trust/ca-bundle.crt`
+file and reload when the file content has been modified.
+
+### Risks and Mitigations
+
+N/A.
+
+### Drawbacks
+
+When `ingress-ca-bundle` ConfigMap contents made available to OpenShift Router as a file
+though volume mount is updated, certain [delay](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#mounted-configmaps-are-updated-automatically) 
+is expected for the new changes to be propagated to the pod.
+
+## Design Details
+
+### Open Questions [optional]
+
+N/A.
+
+### Test Plan
+
+- Create routes with reencrypt termination type with and with destinationCA.
+- Upgrade/downgrade testing.
+- Sufficient time for feedback.
+- Available by default.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Conduct load testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+#### Removing a deprecated feature
+N/A.
+
+### Upgrade / Downgrade Strategy
+
+On upgrade:
+- OpenShift Ingress Operator would have the functionality to look for `admin-ca-bundle`
+  in `openshift-config` namespace and would create `ingress-ca-bundle` in
+  `openshift-ingress` namespace and OpenShift Router would refer to the new CA bundle content.
+
+On downgrade:
+- OpenShift Ingress Operator created `ingress-ca-bundle` in `openshift-ingress` namespace
+  might still exist, but it will not be used by the operator or the router. And the
+  routes created post upgrade without a destinationCA will become inaccessible.
```

Comment:
Do the routes become inaccessible?  They should still use the default CA after downgrade, as before.

Reply:
Routes which relied on the OpenShift service CA(default CA prior to upgrade) will continue to be accessible, but those routes which relied on the custom admin supplied CA certificate will become inaccessible, until user configures the destinationCACertificate for such routes.

Reply:
Okay, please clarify this in the EP.

Reply:
comment incorporated.